src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -404,22 +407,24 @@ void ConnectionHandlerImpl::OnSessionStartedCallback(
 #endif  // ENABLE_SECURITY
                         << " service " << static_cast<int>(service_type)
                         << " for session " << static_cast<int>(session_id));
-      NotifySessionStartedFailure(connection_handle, session_id, is_protected);
+      protocol_handler_->NotifySessionStartedResult(context, empty);
       return;
     }
-    new_session_id = session_id;
-    hash_id = protocol_handler::HASH_ID_NOT_SUPPORTED;
+    context.new_session_id_ = session_id;
+    context.hash_id_ = protocol_handler::HASH_ID_NOT_SUPPORTED;
   }
   sync_primitives::AutoReadLock read_lock(connection_handler_observer_lock_);
   if (connection_handler_observer_) {
-    const uint32_t session_key = KeyFromPair(connection_handle, new_session_id);
+    const uint32_t session_key =
+        KeyFromPair(connection_handle, context.new_session_id_);
+
+    uint32_t app_id = 0;
+    GetDataOnSessionKey(session_key, &app_id, NULL, NULL);
+    if (app_id) {
```

Comment:
@AKalinich-Luxoft avoid implicit casting int to bool

Reply:
@LuxoftAKutsan ok, updated in https://github.com/smartdevicelink/sdl_core/pull/1971/commits/a87c65251e81cb914deaa1f0fe1716175fe1270d